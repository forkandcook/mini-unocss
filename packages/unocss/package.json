{
  "name": "unocss",
  "version": "0.1.0-beta.2",
  "description": "",
  "keywords": [],
  "homepage": "https://github.com/antfu/unocss#readme",
  "bugs": {
    "url": "https://github.com/antfu/unocss/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/antfu/unocss.git"
  },
  "funding": "https://github.com/sponsors/antfu",
  "license": "MIT",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "sideEffects": false,
  "exports": {
    ".": {
      "require": "./dist/index.js",
      "import": "./dist/index.mjs"
    },
    "./vite": {
      "require": "./dist/vite.js",
      "import": "./dist/vite.mjs"
    }
  },
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "types": "index.d.ts",
  "files": [
    "dist",
    "*.d.ts"
  ],
  "scripts": {
    "build": "tsup",
    "dev": "tsup --watch src",
    "prepublishOnly": "nr build"
  },
  "dependencies": {
    "@unocss/core": "workspace:*",
    "@unocss/preset-attributify": "workspace:*",
    "@unocss/preset-uno": "workspace:*",
    "@unocss/vite": "workspace:*"
  },
  "devDependencies": {
    "nanoevents": "^6.0.2"
  }
}
