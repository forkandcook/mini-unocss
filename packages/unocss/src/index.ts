export * from '@unocss/core'
export { default as presetUno } from '@unocss/preset-uno'
export { default as presetAttributify } from '@unocss/preset-attributify'
