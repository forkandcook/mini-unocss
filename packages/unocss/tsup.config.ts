import { Options } from 'tsup'

const config: Options = {
  splitting: true,
  format: ['esm', 'cjs'],
  entryPoints: [
    'src/*.ts',
  ],
  clean: true,
  dts: true,
}

export default config
