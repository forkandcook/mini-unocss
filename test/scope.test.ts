import { createGenerator } from 'unocss'

export const fixture = [
  'pl-10px',
  'hover:p-4',
  '!p-5px',
  '!hover:px-10',
  'hover:!p-10',
  'flex',
  'dark:text-xl',
  'dark:hover:text-xl',
  'sm:text-red-100',
  'sm:text-red-200/10',
  'md:!hidden',
].join(' ')

const uno = createGenerator()

test('scope', async() => {
  const { css } = await uno.generate(fixture, '', '.foo-scope')
  expect(css).toMatchSnapshot()
})
