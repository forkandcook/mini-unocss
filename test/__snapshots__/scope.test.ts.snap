// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scope 1`] = `""`;
